code/modules/reagents/chemistry/reagents/fermi_reagents.dm

```diff
@@ -0,0 +1,157 @@
+//				MISC FERMICHEM CHEMS FOR SPECIFIC INTERACTIONS ONLY
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+/datum/reagent/acidvapour
+	name = "Acid vapour"
+	description = "Someone didn't do like an otter, and add acid to water."
+	taste_description = "acid burns"
+	color = "#FFFFFF"
+	pH = 0
+	can_synth = FALSE
+
+/datum/reagent/acidvapour/reaction_mob(mob/living/carbon/C, method)
+	var/target = C.get_bodypart(BODY_ZONE_CHEST)
+	var/acidstr
+	if(!C.reagents.pH || C.reagents.pH >5)
+		acidstr = 3
+	else
+		acidstr = ((5-C.reagents.pH)*2) //runtime - null.pH ?
+	C.adjustFireLoss(acidstr/2,0)
+	if((method==VAPOR) && (!C.wear_mask))
+		if(prob(20))
+			to_chat(C, "<span class='warning'>You can feel an intense burning sensation in your lungs!</b></span>")
+		C.adjustOrganLoss(ORGAN_SLOT_LUNGS, -2)
+		C.apply_damage(acidstr/5, BURN, target)
+	C.acid_act(acidstr, volume)
+	..()
+
+/datum/reagent/acidvapour/reaction_obj(obj/O, reac_volume)
+	if(ismob(O.loc)) //handled in human acid_act()
+		return
+	if((holder.pH > 5) || (volume < 0.1)) //Shouldn't happen, but just in case
+		return
+	reac_volume = round(volume,0.1)
+	var/acidstr = (5-holder.pH)*2 //(max is 10)
+	O.acid_act(acidstr, volume)
+	..()
+
+/datum/reagent/acidvapour/reaction_turf(turf/T, reac_volume)
+	if (!istype(T))
+		return
+	reac_volume = round(volume,0.1)
+	var/acidstr = (5-holder.pH)
+	T.acid_act(acidstr, volume)
+	..()
+
+/datum/reagent/acidic_buffer
+	name = "Acidic buffer"
+	description = "This reagent will consume itself and move the pH of a beaker towards acidity when added to another."
+	color = "#fbc314"
+	pH = 0
+	can_synth = TRUE
+
+//Consumes self on addition and shifts pH
+/datum/reagent/acidic_buffer/on_new(datapH)
+	if(holder.has_reagent("stabilizing_agent"))
+		return ..()
+	data = datapH
+	if(LAZYLEN(holder.reagent_list) == 1)
+		return ..()
+	holder.pH = ((holder.pH * holder.total_volume)+(pH * (volume)))/(holder.total_volume + (volume))
+	var/list/seen = viewers(5, get_turf(holder))
+	for(var/mob/M in seen)
+		to_chat(M, "<span class='warning'>The beaker fizzes as the pH changes!</b></span>")
+	playsound(get_turf(holder.my_atom), 'sound/FermiChem/bufferadd.ogg', 50, 1)
+	holder.remove_reagent(type, volume, ignore_pH = TRUE)
+	..()
+
+/datum/reagent/basic_buffer
+	name = "Basic buffer"
+	description = "This reagent will consume itself and move the pH of a beaker towards alkalinity when added to another."
+	color = "#3853a4"
+	pH = 14
+	can_synth = TRUE
+
+/datum/reagent/basic_buffer/on_new(datapH)
+	data = datapH
+	if(holder.has_reagent("stabilizing_agent"))
+		return ..()
+	if(LAZYLEN(holder.reagent_list) == 1)
+		return ..()
+	holder.pH = ((holder.pH * holder.total_volume)+(pH * (volume)))/(holder.total_volume + (volume))
+	var/list/seen = viewers(5, get_turf(holder))
+	for(var/mob/M in seen)
+		to_chat(M, "<span class='warning'>The beaker froths as the pH changes!</b></span>")
```

Comment:
Just use visible_message...
`holder.visible_message("<span class='warning'>The beaker froths as the pH changes!</b></span>", vision_distance = 5)`